<launch>

  <arg name="UAV_NAME" default="$(env UAV_NAME)"/>
  <group ns="laserscan_clusters">
    <node name="laserscan_clusters$(arg UAV_NAME)" pkg="laserscan_clusters" type="laserscan_clusters" output="screen"/>
      <rosparam param="obstacles_size"> 0.15 </rosparam>
        <!-- <rosparam param="obstacles_x">
          [-10.0,-9.0,-8.0,-7.0,-5.0,-2.0,-1.0,0.0]
        </rosparam>
        -->
        <!-- <rosparam param="obstacles_y">
          [-35.0,-35.0,-35.0,-35.0,-35.0,-35.0,-35.0,-35.0]
        </rosparam>
        -->
        
        <rosparam param="obstacles_x">
          [-7.0,4.978714499614236, 5.991123934684271, -2.4451078406110183,
          -8.94182644012929, -6.833339293332095, -2.7021949610235936,
          -6.311216361770821, 8.424477213128643, 1.7406538113099224,
          -3.0986796550847746, 4.048699023259182, 2.256339271780048,
          -3.4128453742114733, -9.782024590183884, -0.07589890573841451,
          2.033824399666184, -1.202623062200141, -9.530238683572918,
          3.2252720772109367, -8.532570752008898, 7.451667919701201,
          6.759176842599938, 6.372601593435256, 6.078954907882618,
          -3.7652054132236596, -2.3366175266689337, -5.240814369865827,
          -6.86953121021975, 5.314576475349412, 1.9949351669505138,
          2.4054004943573446, -2.164403481010442, -5.256141616607013,
          4.050337109803619, -8.039803403116604, 6.53692369406361,
          -6.325117895445576, -5.431818249039702, 3.4967240838844944,
          -6.161060350040531, -5.936621577006234, 6.452808081877027,
          5.0448584911141445, -6.788195169202867, -3.3667809505370876,
          -7.812946334569557, 5.453332966482474, -8.233503283064161,
          3.996946272929149, -6.533192222872108, 3.346031257296376,
          4.609769561435076, -4.102266804420567, -9.239724184582306,
          5.062979565692494, -8.46679265440345, 6.785063415863849,
          -6.9136900059428115, -9.515619799025375, 1.37430252144574,
          -9.811273400114642, 8.778423316352718, -9.489704643487073,
          7.121846018590897, 1.948448519443458, 0.20980230319064752,
          9.887956552624356, -3.9961848839104004, 9.298168869234726,
          -1.8811626961412191, 6.037846672124221, -8.293094694865779,
          4.329716612342896, 9.070116459639152, 0.029946047316375513,
          8.692504024945308, 2.5747936460598257, 7.721806453164806,
          0.5153918695517916, 7.712038631421024, -1.8719210627802276,
          7.882226435755854, -7.777695943720477, -0.22957386110213207,
          5.767344254417836, -9.315121620699712, 1.5215714029890464,
          9.67333200581401, -5.95925482783586, -3.5607075455394384,
          9.873566564723067, 9.349661773478896, -0.4599945528053322,
          0.19517777705514483, -4.546830180104873, 8.138437649609259,
          -1.263868165952191, 2.917390628667496, -8.270857685175518,
          -1.4440038450072414, 9.189890789196042, -0.8357724311499517,
          2.2109766520321, -9.415325883992757, 8.958710496049342,
          7.738368617682756, 2.7247749931387215, -4.107710496532826,
          -5.2471390916816585, -7.571667529658095, -8.382240306621753,
          3.9288068757413868, -1.5923305794440168, 1.0003117449165817,
          -0.27820008479648983, 4.139744049847778, -2.2015889947925205,
          3.786954949478652, -3.8472750947224688, -2.7621185822086876,
          0.45020069925954864, -9.603648722056459, -5.272093550555708,
          5.809604046192867, 5.837306868932771, -0.8991284359071514,
          -1.7448269032395096, 9.298006355387919, 0.547654209491748,
          1.9357899142786508, -0.39875951692812883, 5.554362423679578,
          3.5280873738798455, 5.954330986243654, 5.577743237301291,
          -1.4869153812902773, 3.826348782827317, -6.322189970674891,
          9.36785978999626, -3.738508238258884, 6.239903726350629,
          -9.955735979296012, -6.260156250723523, 3.4348861086854185,
          0.20150481650860463, -8.344816413498863, 8.27454330961816,
          5.248556900234611, 0.207207571551697, -9.386430855961299,
          -5.289619494102745, -9.710077610489055, -2.8781080502234495,
          9.681225450206686, -6.902185727835263, 3.2317818190519443,
          -9.818422569455613, -7.7780727344164635, -5.906544134485454,
          9.105120035697567, 2.5073238078430364, -4.425928648467892,
          -2.710885905508018, -9.477878251033706, 0.6473694311742229,
          2.9875415272341, 4.556044256013195, -5.986126856779661,
          -0.17297628035532142, -5.165114233217509, -9.031644416536675,
          -6.82707644396241, 6.418078603114065, -3.0552734634374197,
          3.876182168784318, -5.364995818462555, -6.200242803397902,
          5.081156280005315, 4.199982579245569, 4.970023064640321,
          8.429945672621905, -7.975887932677481, -3.6796315573275518,
          0.029695896006519007, -0.7915425465907582, 6.861234114664768,
          2.5781337964254014, 7.114478757774108, -0.8787487275827033,
          4.426821864473759, 0.1854721036444893, 1.8202113018590929,
          -2.2080903085248904, 4.06146862408556, -0.6375679208144618,
          -7.979150356997106, 7.375721209623855, -7.495574542428017,
          -7.262143049225891, 8.507712286373295]
        </rosparam>
        <rosparam param="obstacles_y">
          [-30.8,7.737485315062024, -11.68204527395028, 3.4758799157498785,
          23.405172885170188, 28.54664552934429, -18.076776853357046,
          19.518908742612354, -31.438391911462425, -27.69696840868844,
          -23.56175084336077, 0.5039641399427168, 3.833862652370989,
          26.596898243534575, -13.672207744834466, -26.014022021523928,
          26.919806904264817, -0.15851230061879562, 8.189094487003949,
          24.61083729824589, -11.578415134746187, -27.83382911988523,
          -0.5853252505544617, -21.284324209508235, -3.3629025285856287,
          0.37027973878010556, -10.79009570539148, -0.39325815028558253,
          -8.554934547009559, 12.522661846122062, 0.6211608692854398,
          -18.598075214021968, -9.085918508069618, 2.724857489558879,
          -8.499926183722614, 20.298734882046922, -18.503979382695213,
          -6.274786407493615, 26.328928816438236, 29.582074281921187,
          -23.755653419464977, 14.24151474881203, 21.6570518740452,
          -0.6884587510824218, 11.474262837707244, -16.17484100516166,
          10.004660328760771, -30.323332325445833, 16.7269017703085,
          13.873253780141816, 30.94232210913885, -29.072885182049575,
          -2.8241841454922323, 28.62073206709406, -9.701016443428394,
          -19.98179878318284, -21.389739500570784, 28.155757184386694,
          -11.885805956020576, 19.10310929870935, -4.449397156034003,
          -26.954495187239104, 5.034156845741585, 25.41958111665671,
          9.338462392433321, 11.659246427710208, -24.46846266083064,
          -30.661675008492992, 7.909826367722744, -15.974032855545893,
          5.706185328408843, 1.8279301897442437, -7.552964452319323,
          -11.958744084442841, -6.292971162247298, -1.5400069570550414,
          -9.87657149241224, 30.98537894682694, 25.780995192354553,
          23.56689436595692, -12.022278895296466, -15.121937167933645,
          30.668984081879145, -3.1533380644335125, 28.99966049232161,
          26.075420485834087, 0.10538115814506455, -9.462440845971763,
          3.5359935730496233, -1.886970879779426, 18.68667880568627,
          9.604911628862254, 25.278283936015704, -19.15055176456955,
          31.14197935493398, -20.229894192748823, -4.885216233339939,
          27.64053384471523, -21.204082434566402, -16.602401949166257,
          14.645254761510962, 19.639698297814547, 11.441671763909063,
          14.022048042023158, -2.973385475389648, 2.1013736917850707,
          -29.876106800618217, 19.697061562033696, -3.762168793222216,
          -8.27852903049444, 22.3689112705486, -30.824338039770403,
          -17.635530649051688, 23.98194617959873, 16.694170318112903,
          -12.213633906217673, 15.409516725429185, 8.123260729360567,
          -19.185517681319027, 15.648516222689942, 31.696573415903963,
          -20.622568631512976, 21.441856985924744, -17.42063351512762,
          -13.499818773360836, 20.141071535176863, -16.750361892359173,
          -2.476000489622116, -22.8173973101812, 25.144948290645075,
          -12.295565556979206, 7.920370856375925, 4.8895089388850295,
          -10.479039236536131, 16.734083192258197, -9.333487264303898,
          -7.213991486658635, -15.833008009114415, -13.298393999994481,
          15.704029115302383, -5.226315862549299, -15.878690383596513,
          5.496509138010417, -3.6130577291128105, -25.747357395637195,
          -7.4221724185902715, -1.5000492303524595, 16.860419765954603,
          -22.344855666853753, -22.583935148779148, 11.456107836885174,
          22.296327893924555, 13.780081643002816, 25.067600278304504,
          -2.276518476500435, -14.76349160542734, 6.875949129506317,
          -5.334826262540211, 1.1909058405275204, -30.64833173122878,
          -25.23103360417526, 21.488658237989434, -14.155532653447892,
          -30.20966173057834, -18.365528257415548, 1.5554956754568536,
          -31.137352835247505, 18.539516814974697, 15.785826877070775,
          9.652221152784875, -27.467017198691416, 30.617299893777385,
          -19.039504759978072, -5.395883418441407, -27.740705479101685,
          4.960616523692323, -15.600548757781787, 7.550516693388701,
          3.4525219059247974, 23.37518462990088, -26.721463279787216,
          -13.48400056616628, 13.031195513275847, 14.017914108687343,
          20.741557672126582, 3.1743433059070014, 23.263366990307304,
          17.68307035529785, -24.741888972725576, -4.286659042976019,
          27.825148621987395, 4.323445736339636, -0.8747781308305704,
          9.935422310363748, 9.955458898943071, 16.22824713230795,
          -25.158618422269782, -9.022195849478642, 6.712705781203439,
          3.3669914696790073, -21.38161906896321]
        </rosparam>

      <!-- <rosparam param="obstacles_x">
        [0.0,-10.0]
      </rosparam>
        -->
        <!-- <rosparam param="obstacles_y">
          [-15.0,-15.0]
        </rosparam>
          -->
          <rosparam param="clustering/tolerance">
            0.5
          </rosparam>
          <rosparam param="clustering/min_size">
            2
          </rosparam>
          <rosparam param="clustering/max_size">
            300
          </rosparam>
          <rosparam param="simulation">
            true
          </rosparam>
        </group>
      </launch>
